src/mlx/warnings_checker.py

```diff
@@ -3,11 +3,11 @@
 
 import abc
 import re
-from junitparser import JUnitXml, Failure, Error
 from xml.etree.ElementTree import ParseError
 
+from junitparser import Error, Failure, JUnitXml
 
-DOXYGEN_WARNING_REGEX = r"(?:((?:[/.]|[A-Za-z]).+?):(-?\d+):\s*([Ww]arning|[Ee]rror)|<.+>:-?\d+(?::\s*([Ww]arning|[Ee]rror))?): (.+(?:(?!\s*(?:[Nn]otice|[Ww]arning|[Ee]rror): )[^/<\n][^:\n][^/\n].+)*)|\s*([Nn]otice|[Ww]arning|[Ee]rror): (.+)\n?"
+DOXYGEN_WARNING_REGEX = r"(?:((?:[/.]|[A-Za-z]).+?):(-?\d+):\s*([Ww]arning|[Ee]rror)|<.+> :-?\d+(?::\s*([Ww]arning|[Ee]rror))?): ([^notes].+(?:(?!\s*(?:[Nn]otice|[Ww]arning|[Ee]rror): )[^/<\n][^:\n][^/\n].+))|\s([Nn]otice|[Ww]arning|[Ee]rror): (.+)\n?"
```

Comment:
Did you mean to add a whitespace and remove `*` twice in this regex?

Reply:
While the use of `^` in `[^notes]` passes the unit test, it is not a good solution. I suggest making use of `(?!...)` _(negative lookahead assertion)_ to exclude _notes_ as a string instead of excluding all characters in _notes_.

Reply:
113a3ce85a7bf5e1c489217befd7923100112e2e

4f17d59e8a197ecd3b8e581fa23c288ab3c00c01